frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -184,6 +246,20 @@ pub mod pallet {
 			})
 		}
 
+		#[transactional]
+		fn set_pool_id_for_asset(
+			asset_id: T::AssetId,
+			pool_id: T::PoolId,
+		) -> Result<(), DispatchError> {
+			if Pools::<T>::contains_key(asset_id) {
+				Pools::<T>::mutate(asset_id, |_| pool_id);
+				Ok(())
```

Comment:
These `Ok(())`'s can be moved out of the if else block since both conditions result in it